mgapi/handler.py

```diff
@@ -0,0 +1,109 @@
+""" Corpus Fetcher
+
+ Loads corpus data from MemeGenerator.net API
+ """
+
+import urllib.request
+import json
+
+
+class MemeGeneratorApiHandler:
+    """ Fetches mgapi data via API """
+    def __init__(self, username, password):
+        """ initializes l/p """
+
+        self._username = username
+        self._password = password
+
+    def _get_request_url(self, action, params=''):
+        """ String builder for API endpoints """
+        url = 'http://{}/{}'.format(
+            'version1.api.memegenerator.net',
+            action
+        )
+
+        for i, param in enumerate(params):
+            url = self._add_url_param(url, param, params[param])
+
+        return url
+
+    def _add_url_param(self, url, name, val):
+        """ adds a parameter to the URL string """
+        delim = '&' if '?' in url else '?'
+        param = '{}{}={}'.format(delim, name, val)
+        return url + param
+
+    def _load_url_as_json(self, url):
+        """ Loads a URL, returns JSON data as an object
+
+        :return json_data
+        """
+        with urllib.request.urlopen(url) as text:
+            data_str = '\n'.join([line.decode('UTF-8') for line in text])
+
+        data_json = json.loads(data_str)
```

Comment:
I recommend using the [requests](http://docs.python-requests.org/en/master/) Python library whenever doing any HTTP getting. Even the [standard lib docs for urllib.request suggest you just use requests](https://docs.python.org/3/library/urllib.request.html).

Turns out it already has implemented three of your functions here for you. You can jump straight to using [requests.get()](http://docs.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls) with parameters. Then you don't need `_get_request_url()` or `_add_url_param()`.